### PR TITLE
Issue 173. Check maxIterations before backoff calculation.

### PR DESCRIPTION
**Pull request by alex-pumpkin**
https://github.com/reactor/reactor-addons/issues/173

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@alex-pumpkin Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-addons&pullRequestId=174)!

[Click here](https://cla.pivotal.io/sync/reactor?repositoryId=reactor/reactor-addons&pullRequestId=174) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@alex-pumpkin Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-addons&pullRequestId=174)!

File: reactor-extra/src/test/java/reactor/retry/RepeatTests.java [hunk 1]
```diff
@@ -399,7 +399,7 @@ public void functionReuseInParallel() throws Exception {
 			for (int j = 1; j <= range; j++)
 				values[i * range + j - 1] = j;
 		}
-		RetryTestUtils.<Long>testReuseInParallel(2, 20,
+		RetryTestUtils.<Long>testReuseInParallel(2, 19,
```
Comment (simonbasle) [inline reactor-extra/src/test/java/reactor/retry/RepeatTests.java]: can you revert back to `20` and change the assertion in `RetryTestUtils` instead to be 
```java
	for (Integer count : backoffCounts.values()) {
			//backoff not invoked anymore when maxIteration reached
			assertEquals(repeatCount, count.intValue());
		}
```

File: reactor-extra/src/test/java/reactor/retry/RetryTests.java [hunk 2]
```diff
@@ -336,7 +336,7 @@ public void functionReuseInParallel() throws Exception {
 			for (int j = 1; j <= range; j++)
 				values[i * range + j - 1] = j;
 		}
-		RetryTestUtils.<Throwable>testReuseInParallel(2, 20,
+		RetryTestUtils.<Throwable>testReuseInParallel(2, 19,
```
Comment (simonbasle) [inline reactor-extra/src/test/java/reactor/retry/RetryTests.java]: same as above

**Comment by alex-pumpkin** [comment]
Reopened for 3.1.x with test fixes.

**Pull request closed without merging**